sky/provision/kubernetes/network_utils.py

```diff
@@ -0,0 +1,170 @@
+import enum
+import os
+from typing import Dict, List, Tuple, Union
+
+import jinja2
+import yaml
+
+import sky
+from sky import exceptions
+from sky.adaptors import kubernetes
+
+_INGRESS_TEMPLATE_NAME = 'kubernetes-ingress.yml.j2'
+_LOADBALANCER_TEMPLATE_NAME = 'kubernetes-loadbalancer.yml.j2'
+
+
+class KubernetesPortMode(enum.Enum):
+    """Enum for the different types of modes supported for opening
+    ports on Kubernetes.
+    """
+    INGRESS = 'ingress'
+    LOADBALANCER = 'loadbalancer'
```

Comment:
This is fantastic! I love the flexibility this provides. Would be nice to document this in `docs/source/reference/config.rst` :)

Reply:
For sure, will update.